src/autocomplete/lib/autocomplete-parse-result.ts

```diff
@@ -0,0 +1,594 @@
+    | 'limitClause';
+
+export interface IdentifierLocation {
+    identifier?: string;
+    type: IdentifierLocationType;
+    alias?: string;
+    source?: string;
+    location: ParsedLocation;
+    function?: string;
+    missing?: boolean;
+    value?: string;
+    active?: boolean;
+    tables?: ParsedTable[];
+    colRef?: DetailedColumnReference;
+    argumentPosition?: number;
+    identifierChain?: IdentifierChainEntry[];
+    expression?: AwaitedTokenExpression;
+    parentLocation?: ParsedLocation;
+    path?: string;
+    qualified?: boolean;
+    columns?: ColumnDetails[];
+    subquery?: boolean;
+    suffix?: string;
+    linked?: boolean;
+    firstInChain?: boolean;
+    target?: string;
+}
+
+export interface DetailedColumnReference {
+    identifierChain: IdentifierChainEntry[];
+    tables?: ParsedTable[];
+    linked?: boolean;
+    owner?: string;
+}
+
+export interface ColumnAliasDetails {
+    name: string;
+    udfRef?: string;
+    types: string[];
+}
+
+export interface ColumnDetails {
+    type?: string | string[];
+    identifierChain?: IdentifierChainEntry[];
+    location?: ParsedLocation;
+    alias?: string;
+    subQuery?: string;
+    udfRef?: string;
+}
+
+export interface CommonPopularSuggestion {
+    tables?: ParsedTable[];
+    tablePrimaries?: ParsedTable[];
+    prefix?: string;
+    linked?: boolean;
+}
+
+export type OrderBysSuggestion = CommonPopularSuggestion;
+export type GroupBysSuggestion = CommonPopularSuggestion;
+
+export interface HdfsSuggestion {
+    path: string;
+}
+
+export interface AutocompleteParseResult {
+    colRef?: DetailedColumnReference;
+    commonTableExpressions?: IdentifierLocation[];
+    locations?: IdentifierLocation[];
+    lowerCase?: boolean;
+    subQueries?: SubQuery[];
+    suggestAggregateFunctions?: AggregateFunctionsSuggestion;
+    suggestAnalyticFunctions?: boolean;
+    suggestColRefKeywords?: ColRefKeywordsSuggestion;
+    suggestColumnAliases?: ColumnAliasDetails[];
+    suggestColumns?: ColumnSuggestion;
+    suggestCommonTableExpressions?: IdentifierSuggestion[];
+    suggestDatabases?: DatabasesSuggestion;
+    suggestFilters?: FiltersSuggestion;
+    suggestFunctions?: FunctionsSuggestion;
+    suggestGroupBys?: GroupBysSuggestion;
+    suggestHdfs?: HdfsSuggestion;
+    suggestJoins?: JoinsSuggestion;
+    suggestJoinConditions?: JoinConditionsSuggestion;
+    suggestIdentifiers?: IdentifierSuggestion[];
+    suggestKeyValues?: KeyValuesSuggestion;
+    suggestKeywords?: WeightedKeywordSuggestion[];
+    suggestOrderBys?: OrderBysSuggestion;
+    suggestSetOptions?: boolean;
+    suggestTables?: TablesSuggestion;
+    suggestValues?: ValuesSuggestion;
+    udfArgument?: UdfArgumentPosition;
+    useDatabase?: string;
+    error?: {
+        expected?: string[];
+        recoverable?: boolean;
+    };
+    errors?: ErrorLocation[];
+    definitions?: IdentifierLocation[];
+    suggestTemplates?: boolean;
+    suggestEngines?: EnginesSuggestion;
+}
```

Comment:
Let's reorder those fields as it was. People (and us) will look at this interface a lot, so all the important stuff should be first. Also if possible, please move this interface to the top of the file, because it's the most important interface and should be easily found

```suggestion
export interface AutocompleteParseResult {
    // Meta
    locations?: IdentifierLocation[];
    errors?: ErrorLocation[];
    
    // Suggestions
    suggestAggregateFunctions?: AggregateFunctionsSuggestion;
    suggestAnalyticFunctions?: boolean;
    suggestColRefKeywords?: ColRefKeywordsSuggestion;
    suggestColumnAliases?: ColumnAliasDetails[];
    suggestColumns?: ColumnSuggestion;
    suggestCommonTableExpressions?: IdentifierSuggestion[];
    suggestDatabases?: DatabasesSuggestion;
    suggestFilters?: FiltersSuggestion;
    suggestFunctions?: FunctionsSuggestion;
    suggestGroupBys?: GroupBysSuggestion;
    suggestHdfs?: HdfsSuggestion;
    suggestJoins?: JoinsSuggestion;
    suggestJoinConditions?: JoinConditionsSuggestion;
    suggestIdentifiers?: IdentifierSuggestion[];
    suggestKeyValues?: KeyValuesSuggestion;
    suggestKeywords?: WeightedKeywordSuggestion[];
    suggestOrderBys?: OrderBysSuggestion;
    suggestSetOptions?: boolean;
    suggestTables?: TablesSuggestion;
    suggestValues?: ValuesSuggestion;
    suggestTemplates?: boolean;
    suggestEngines?: EnginesSuggestion;
    
    // Other
    subQueries?: SubQuery[];
    commonTableExpressions?: IdentifierLocation[];
    useDatabase?: string;
    udfArgument?: UdfArgumentPosition;
    error?: {
        expected?: string[];
        recoverable?: boolean;
    };
    definitions?: IdentifierLocation[];
    colRef?: DetailedColumnReference;
    lowerCase?: boolean;
}
```